read.c

```diff
@@ -213,7 +216,7 @@ static int string2ll(const char *s, size_t slen, long long *value) {
     if (negative) {
         if (v > ((unsigned long long)(-(LLONG_MIN+1))+1)) /* Overflow. */
             return REDIS_ERR;
-        if (value != NULL) *value = -v;
+        if (value != NULL) *value = -(long long)v;
```

Comment:
This is a bug!.  The negation of an unsigned is a no-op.  It must first be cast to a signed int.